hoodie-client/src/main/java/com/uber/hoodie/io/compact/HoodieRealtimeTableCompactor.java

```diff
@@ -0,0 +1,169 @@
+            jsc.parallelize(operations, operations.size()).map(
+                (Function<CompactionOperation, Iterator<List<WriteStatus>>>) compactionOperation -> executeCompaction(
+                    metaClient, config, compactionOperation, compactionCommit)).flatMap(
+                (FlatMapFunction<Iterator<List<WriteStatus>>, WriteStatus>) listIterator -> {
+                    List<List<WriteStatus>> collected = IteratorUtils.toList(listIterator);
+                    return collected.stream().flatMap(List::stream).collect(Collectors.toList());
+                }).mapToPair(new PairFunction<WriteStatus, String, HoodieWriteStat>() {
+                @Override
+                public Tuple2<String, HoodieWriteStat> call(WriteStatus writeStatus)
+                    throws Exception {
+                    return new Tuple2<>(writeStatus.getPartitionPath(), writeStatus.getStat());
+                }
+            }).collect();
+
+        HoodieCompactionMetadata metadata = new HoodieCompactionMetadata();
+        for (Tuple2<String, HoodieWriteStat> stat : updateStatusMap) {
+            metadata.addWriteStat(stat._1(), stat._2());
+        }
+        log.info("Compaction finished with result " + metadata);
+
+        //noinspection ConstantConditions
+        if (isCompactionSucceeded(metadata)) {
+            log.info("Compaction succeeded " + compactionCommit);
+            commitCompaction(compactionCommit, metaClient, metadata);
+        } else {
+            log.info("Compaction failed " + compactionCommit);
+        }
+        return metadata;
+    }
+
+    private boolean isCompactionSucceeded(HoodieCompactionMetadata result) {
+        //TODO figure out a success factor for a compaction
+        return true;
+    }
+
+    private Iterator<List<WriteStatus>> executeCompaction(HoodieTableMetaClient metaClient,
+        HoodieWriteConfig config, CompactionOperation operation, String commitTime)
+        throws IOException {
+        FileSystem fs = FSUtils.getFs();
+        Schema schema =
+            HoodieAvroUtils.addMetadataFields(new Schema.Parser().parse(config.getSchema()));
+
+        log.info("Compacting base " + operation.getDataFilePath() + " with delta files " + operation
+            .getDeltaFilePaths() + " for commit " + commitTime);
+        // TODO - FIX THIS
+        // 1. Reads the entire avro file. Always only specific blocks should be read from the avro file (failure recover).
+        // Load all the delta commits since the last compaction commit and get all the blocks to be loaded and load it using CompositeAvroLogReader
+        // Since a DeltaCommit is not defined yet, reading all the records. revisit this soon.
+
+        // 2. naively loads all the delta records in memory to merge it,
+        // since we only need a iterator, we could implement a lazy iterator to load from one delta file at a time
+        List<HoodieRecord<HoodieAvroPayload>> readDeltaFilesInMemory =
```

Comment:
We should probably load the base files into memory and run the delta files through it? the other way around? 

May be just add a TODO, its okay for now..